src/ol/interaction/Snap.js

```diff
@@ -613,17 +622,4 @@ class Snap extends PointerInteraction {
 }
 
 
-/**
- * Sort segments by distance, helper function
- * @param {SegmentData} a The first segment data.
- * @param {SegmentData} b The second segment data.
- * @return {number} The difference in distance.
- * @this {Snap}
- */
-function sortByDistance(a, b) {
-  const deltaA = squaredDistanceToSegment(this.pixelCoordinate_, a.segment);
-  const deltaB = squaredDistanceToSegment(this.pixelCoordinate_, b.segment);
-  return deltaA - deltaB;
-}
```

Comment:
Previously we were sorting the full list of segments but only using the first.  Now we go through the list once to get the closest instead of sorting the whole list.